### PR TITLE
[RLlib] ES + ARS config objects.

### PR DESCRIPTION
**Pull request by sven1977**
ES + ARS config objects.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/ars/ars.py [hunk 1]
```diff
@@ -34,31 +36,122 @@
...
-    },
-})
-# __sphinx_doc_end__
-# fmt: on
+
+class ARSConfig(TrainerConfig):
+    """Defines an ARSTrainer configuration class from which an ARSTrainer can be built.
+
+    Example:
+        >>> config = ARSConfig().training(sgd_stepsize=0.02, report_length=20)\
```
Comment (maxpumperla) [inline rllib/agents/ars/ars.py:44]: Please import `ARSConfig` for both examples here, too. thanks!

File: rllib/agents/es/es.py [hunk 2]
```diff
@@ -33,32 +34,123 @@
...
-    },
-})
-# __sphinx_doc_end__
-# fmt: on
+
+class ESConfig(TrainerConfig):
+    """Defines an ESTrainer configuration class from which an ESTrainer can be built.
+
+    Example:
+        >>> config = ESConfig().training(sgd_stepsize=0.02, report_length=20)\
```
Comment (maxpumperla) [inline rllib/agents/es/es.py:42]: same here

File: rllib/agents/es/es.py [hunk 3]
```diff
@@ -33,32 +34,123 @@
...
+        stepsize: Optional[float] = None,
+        noise_size: Optional[int] = None,
+        report_length: Optional[int] = None,
+        **kwargs,
+    ) -> "ESConfig":
+        """Sets the training related configuration.
+
+        Args:
+            action_noise_std:
+            l2_coeff:
```
Comment (maxpumperla) [inline rllib/agents/es/es.py:119]: documentation seems to be missing for these parameters - I think we need those!

File: rllib/agents/ars/ars.py [hunk 4]
```diff
@@ -34,31 +36,122 @@
...
+        noise_size: Optional[int] = None,
+        eval_prob: Optional[float] = None,
+        report_length: Optional[int] = None,
+        offset: Optional[int] = None,
+        **kwargs,
+    ) -> "ARSConfig":
+        """Sets the training related configuration.
+
+        Args:
+            action_noise_std:
```
Comment (maxpumperla) [inline rllib/agents/ars/ars.py:119]: missing arg docs

File: rllib/agents/ppo/ppo.py [hunk 5]
```diff
@@ -53,7 +53,7 @@ class PPOConfig(TrainerConfig):
     """Defines a PPOTrainer configuration class from which a PPOTrainer can be built.
 
     Example:
-        >>> config = PPOConfig(kl_coeff=0.3).training(gamma=0.9, lr=0.01)\
+        >>> config = PPOConfig().training(gamma=0.9, lr=0.01)\
```
Comment (maxpumperla) [inline rllib/agents/ppo/ppo.py:56]: yep, I caught that one in my PR as well (also still needs the import). I wonder why `doctest` doesn't complain about this in CI, tbh...

**Review by maxpumperla — CHANGES_REQUESTED** [review]
also just minor nits! :D

File: rllib/agents/es/es.py (hunk 3 above)
Comment (sven1977) [reply rllib/agents/es/es.py:119]: Yeah, it never existed in the first place :/ Will try to find out what these do exactly. ...

File: rllib/agents/ars/ars.py (hunk 4 above)
Comment (sven1977) [reply rllib/agents/ars/ars.py:119]: same

**Review by maxpumperla — APPROVED** [review]

**Pull request merged**